### ISSUE TITLE
`data Foo constructor {cs : ts}` notation in `interleaved mutual`

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#5336]
In `interleaved mutual` blocks, I find myself writing comments which `data` some `constructor`s belong to.  For example,
```agda
  constructor  -- _⊆_
    refl-⊆ : sc ⊆ sc
    _▷_  : (τ : sc ⊆ sc') (ws : WkDecls τ ds ds') → (sc ▷ ds) ⊆ (sc' ▷ ds')
    _▷ʳ_ : (τ : sc ⊆ sc') (ds : Decls sc')        → sc        ⊆ (sc' ▷ ds)
```
It would be nice if one could make this formal.  Concrete suggestion:
```agda
  data _⊆_ constructor
    refl-⊆ : sc ⊆ sc
    _▷_  : (τ : sc ⊆ sc') (ws : WkDecls τ ds ds') → (sc ▷ ds) ⊆ (sc' ▷ ds')
    _▷ʳ_ : (τ : sc ⊆ sc') (ds : Decls sc')        → sc        ⊆ (sc' ▷ ds)
```
This could also help Agda sorting constructors into data types in situations like #5334.

**Comment by andreasabel** [comment]
After implementing this in #5350, I noticed that `data...where` is cumulative, so the existing syntax could work:
```agda
  data _⊆_ where
    refl-⊆ : sc ⊆ sc
  ...
  data _⊆_ where
    _▷_  : (τ : sc ⊆ sc') (ws : WkDecls τ ds ds') → (sc ▷ ds) ⊆ (sc' ▷ ds')
    _▷ʳ_ : (τ : sc ⊆ sc') (ds : Decls sc')        → sc        ⊆ (sc' ▷ ds)
```
Test case in #5351.

**Comment by andreasabel** [comment]
Fixed by PR #5351.
